src/utils/requestUtils.ts

```diff
@@ -48,4 +50,27 @@ export namespace requestUtils {
             stream.pipe(fse.createWriteStream(filePath).on('finish', resolve).on('error', reject));
         });
     }
+
+    // tslint:disable-next-line:no-any
+    export async function getFunctionAppMasterKey(resourceId: string, actionContext: IActionContext): Promise<WebSiteManagementModels.HostKeys | undefined> {
+        const slotTreeItem: SlotTreeItemBase | undefined = await ext.tree.findTreeItem(resourceId, { ...actionContext, loadAll: true });
+        return await slotTreeItem?.client.listHostKeys();
+    }
+
+    export async function downloadFunctionAppContent(defaultHostName: string, filePath: string, masterKey: string): Promise<void> {
```

Comment:
No need to copy a bunch of code here. I'd suggest you tweak `downloadFile` like this and then you can just pass in the request options:
```typescript
    export async function downloadFile(requestOptionsOrUrl: string | RequestPrepareOptions, filePath: string): Promise<void> {
        await fse.ensureDir(path.dirname(filePath));
        const request: WebResource = new WebResource();
        request.prepare(typeof requestOptionsOrUrl === 'string' ? { method: 'GET', url: requestOptionsOrUrl } : requestOptionsOrUrl);
```